spec/lib/appsignal/hooks/rake_spec.rb

```diff
@@ -1,72 +1,53 @@
-    context "with transaction" do
-      let!(:transaction) { background_job_transaction }
-      let!(:agent)       { double('Agent', :send_queue => true) }
-      before do
-        transaction.stub(:set_action)
-        transaction.stub(:set_error)
-        transaction.stub(:complete)
-        SecureRandom.stub(:uuid => '123')
-        Appsignal::Transaction.stub(:create => transaction)
-        Appsignal.stub(:active? => true)
-      end
-
-      it "should call the original task" do
-        expect( task ).to receive(:execute_without_appsignal).with('foo')
+    context "without error" do
+      it "creates no transaction" do
+        expect(Appsignal::Transaction).to_not receive(:create)
       end
 
-      it "should not create a transaction" do
-        expect( Appsignal::Transaction ).not_to receive(:create)
+      it "calls the original task" do
+        expect(task).to receive(:execute_without_appsignal).with('foo')
       end
 
-      context "with an exception" do
-        let(:exception) { VerySpecificError.new }
-
-        before do
-          task.stub(:execute_without_appsignal).and_raise(exception)
-        end
+      after { task.execute('foo') }
+    end
 
-        it "should create a transaction" do
-          expect( Appsignal::Transaction ).to receive(:create).with(
-            '123',
-            Appsignal::Transaction::BACKGROUND_JOB,
-            kind_of(Appsignal::Transaction::GenericRequest)
-          )
-        end
+    context "with error" do
+      let(:error) { VerySpecificError.new }
+      let(:transaction) { background_job_transaction }
+      before do
+        task.enhance { raise error }
 
-        it "should set the action on the transaction" do
-          expect( transaction ).to receive(:set_action).with('task:name')
-        end
+        expect(Appsignal::Transaction).to receive(:create).with(
+          kind_of(String),
+          Appsignal::Transaction::BACKGROUND_JOB,
+          kind_of(Appsignal::Transaction::GenericRequest)
+        ).and_return(transaction)
+      end
 
-        it "should add the exception to the transaction" do
-          expect( transaction ).to receive(:set_error).with(exception)
-        end
+      it "sets the action" do
+        expect(transaction).to receive(:set_action).with('task:name')
+      end
 
-        it "should call complete! on the transaction" do
-          expect( transaction ).to receive(:complete)
-        end
+      it "sets the error" do
+        expect(transaction).to receive(:set_error).with(error)
+      end
 
-        it "should stop appsignal" do
-          expect( Appsignal ).to receive(:stop)
-        end
+      it "completes the transaction and stops" do
+        expect(transaction).to receive(:complete).ordered
+        expect(Appsignal).to receive(:stop).with('rake').ordered
       end
-    end
 
-    after { task.execute('foo') rescue VerySpecificError }
+      after { expect { task.execute('foo') }.to raise_error error.class }
```

Comment:
To clarify this a bit, can we expect `VerySpecificError` here? Also, a nitpick, as this has multiple nested blocks, I'd use a full do..end `after` block for a little more readability:

``` ruby
after do
  expect { task.execute('foo') }.to raise_error VerySpecificError
end
```